### PR TITLE
added pathSide property to text on path

### PR DESCRIPTION
**Pull request by melchiar**
Here's my newest implementation of the `side` property for text on a path. Note that I've used the name `side` to be consistent with the related [SVG property](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/side), although SVG `side` is currently supported only in Firefox.

Feel free to test:
https://jsfiddle.net/melchiar/oLq91r2t/

### CODE REVIEW

File: src/shapes/text.class.js [hunk 1]
```diff
@@ -212,6 +212,14 @@
      */
     startOffset:               0,
 
+    /**
+     * Which side of the path the text should be drawn on.
+     * Only used when text has a path
+     * @type {String} 'left|right'
+     * @default
+     */
+    side:               'left',
```
Comment (asturur) [inline src/shapes/text.class.js:221]: for the previous discussion, should we call this `pathSide`?

File: src/shapes/text.class.js [hunk 2]
```diff
@@ -818,6 +832,14 @@
...
       graphemeInfo.renderTop = info.y - startingPoint.y;
       graphemeInfo.angle = info.angle;
+      switch (this.side) {
+        case 'left':
+          graphemeInfo.angle = info.angle;
+          break;
+        case 'right':
+          graphemeInfo.angle = info.angle + Math.PI;
+          break;
+      }
```
Comment (asturur) [inline src/shapes/text.class.js:842]: why in other part of the code we use a ternary and here a switch case? Are there are other values of side we should be aware of? if is just left/rigght we can have 
```
graphemeInfo.angle = info.angle + this.side ===  'right' ? Math.PI : 0;
```

File: src/shapes/text.class.js [hunk 3]
```diff
@@ -818,6 +832,14 @@
       graphemeInfo.renderLeft = info.x - startingPoint.x;
       graphemeInfo.renderTop = info.y - startingPoint.y;
       graphemeInfo.angle = info.angle;
```
Comment (asturur) [inline src/shapes/text.class.js:834]: this seems a leftover line?

Comment (melchiar) [reply src/shapes/text.class.js:834]: oops

File: src/shapes/text.class.js (hunk 2 above)
Comment (melchiar) [reply src/shapes/text.class.js:842]: you're right, that's better

File: src/shapes/text.class.js (hunk 1 above)
Comment (melchiar) [reply src/shapes/text.class.js]: sounds good. I think let's change `startOffset` to `pathStartOffset` as well for the same reason

Comment (asturur) [reply src/shapes/text.class.js]: yes i agree is fresh enough that no one noticed it yet.

**Comment by asturur** [comment]
i'm merging.
Can you follow up with a visual test?

**Pull request merged**